tests/scripts/rm_ml_api_tests.sh

```diff
@@ -0,0 +1,293 @@
+		
+		get_json_array ${current_id}
+		post_json "${rm_ml_post_experiment_json[$exp]}"
+	
+		if [[ "${post}" == valid* ]]; then
+			expected_result_="200"
+			expected_behaviour="RESPONSE_CODE = 200 OK"
+		else
+			expected_result_="^4[0-9][0-9]"
+			expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+		fi
+		
+		actual_result="${http_code}"
+		if [[ "${http_code}" -eq "000" ]]; then
+			expected_log_msg="${rm_ml_error_messages[${exp}]}"
+			if grep -q "${expected_log_msg}" "${TESTS_}/service.log" ; then
+				failed=0 
+			else
+				failed=1
+			fi
+			error_message ${failed}
+		else
+			compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+		fi
+		echo ""
+		
+		# Stop the python servers
+		${SCRIPTS_DIR}/start_python_servers.sh -t | tee -a ${LOG_} ${LOG}
+		
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 2
+		
+		echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}
+		
+	done
+}
+
+# Do a post on experiment_trials for the same id again with "operation: EXP_TRIAL_GENERATE_NEW" and check if experiments have started from the begining
+function operation_generate_new() {
+	flag=0
+	get_json_array ${current_id}
+	post_json "${rm_ml_post_experiment_json[$exp]}"
+	
+	# Post the json with same Id having "operation: EXP_TRIAL_GENERATE_NEW"
+	echo "Post the json with same Id having operation: EXP_TRIAL_GENERATE_NEW" | tee -a ${LOG_} ${LOG}
+	post_json "${rm_ml_post_experiment_json[$exp]}"
+		
+	actual_result="${http_code}"
+	expected_result_="^4[0-9][0-9]"
+	expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+	
+	compare_result ${FUNCNAME} ${expected_result_} "${expected_behaviour}"
+}
+
+# Post the experiment result to python
+# input: Experiment result
+# output: Create the Curl command with given JSON and get the result
+function post_trial() {
+	exp_result='{"id" : "'${current_id}'", "trial_number": 0, "trial_result": "success", "result_value_type": "double", "result_value": 98.78, "operation" : "EXP_TRIAL_RESULT"}'
+	_post_=$(curl -H "Content-Type: application/json" -d "${exp_result}" http://localhost:8085/experiment_trials -w '\n%{http_code}' 2>&1)
+	post_exp_result_cmd='curl -H "Content-Type: application/json" -d '${exp_result}' http://localhost:8085/experiment_trials -w '\n%{http_code}''
+
+	echo "" | tee -a ${LOG_} ${LOG}
+	echo "Command used to post the experiment result= ${post_exp_result_cmd}" | tee -a ${LOG_} ${LOG}
+	echo "" | tee -a ${LOG_} ${LOG}
+
+	echo "${_post_}" >> ${LOG_} ${LOG}
+
+	http_code=$(tail -n1 <<< "${_post_}")
+	response=$(echo -e "${_post_}" | tail -2 | head -1)
+	
+	echo "Response is ${response}" >> ${LOG_} ${LOG}
+}
+
+# Do a post on experiment_trials for the same id again with "operation: EXP_TRIAL_GENERATE_SUBSEQUENT" and check if same experiment continues
+function operation_generate_subsequent() {
+	get_json_array ${current_id}
+	post_json "${rm_ml_post_experiment_json[$exp]}"
```

Comment:
Fetch the trial number after posting the experiment and use it later to compare the subsequent trial number